klib/mbedtls.c

```diff
@@ -296,6 +296,12 @@ void mbedtls_platform_zeroize( void *buf, size_t len )
 
 void *mbedtls_calloc(size_t n, size_t s)
 {
+    /* To maintain the malloc/free interface with mcache, allocations must stay
+       within the range of objcaches and not fall back to parent allocs. */
+    if (s > U64_FROM_BIT(MAX_MCACHE_ORDER)) {
+        tls.rprintf("%s: size %ld exceeds max alloc order\n", __func__, s);
```

Comment:
It should be `n * s` instead of `s` (both in the `if` condition and in the `rprintf` call).
I verified that mbedtls does not exceed the max alloc order when connecting to an HTTPS server (in my test, the maximum amount it allocates at once is 16717 bytes).

Reply:
Ok, I fixed this. Thanks for checking mbedtls allocs.